packages/openapi-fetch/test/middleware/middleware.test.ts

```diff
@@ -505,3 +505,118 @@ test("skips onResponse handlers when response is returned from onRequest", async
 
   expect(onResponseCalled).toBe(false);
 });
+
+test('it should enable a middleware to be added via the "middleware" request option', async () => {
+  let actualRequest = new Request("https://nottherealurl.fake");
+  const client = createObservedClient<paths>({}, async (req) => {
+    actualRequest = new Request(req);
+    return Response.json({});
```

Comment:
Throw in the fetch body? It should never actually be called. See:

https://github.com/Kauabunga/openapi-typescript/blob/7527d1e7502cd1e9621922f028b4736d85f25800/packages/openapi-fetch/test/middleware/middleware.test.ts#L447-L465